### DISCUSSION TITLE
Is there a way to stop attaching to a process when using "-o" option

### DISCUSSION THREAD

**Discussion by EricLin89** [bloomberg/memray#469] [Q&A]
I used command `memray attach <pid> -o <output.bin>` to analyze the process's memory allocation. 
After gathering enough data, I would like to stop the attachment, because otherwise the output file would eventually consume all my disk space. 
But I can't find a way to do this other than stopping the process, which I would rather avoid.
Deleting the output file doesn't work because the file handler is still occupied by the process and the disk won't be freed until the file is closed.

**Comment by godlygeek** [comment]
> After gathering enough data, I would like to stop the attachment

We're working on adding a command to do this right now in #458, so keep an eye out for our next release.

> But I can't find a way to do this other than stopping the process, which I would rather avoid.

In the meantime, calling `memray attach` a second time happens to gracefully stop the tracking that was started by a previous call to `memray attach`, so a hacky way to get what you want with the current Memray version is to re-attach without an output file, stopping tracking to the file and starting tracking with the live-mode TUI - which you can then just quit out of by pressing `q` and all the tracking will be stopped.

If you can't use the TUI because you're in an headless environment with no terminal, you could re-attach and specify an invalid file path as the output file, something like `memray attach <pid> -o /` — that will fail and print an error (because `/` isn't a valid output file), _but_ it will detach the old tracker before it fails.
